MAVProxy/modules/mavproxy_chat/chat_voice_to_text.py

```diff
@@ -53,22 +55,18 @@ def record_audio(self):
             print("chat: failed to connect to microphone")
             return None
 
-        # calculate time recording should stop
-        curr_time = time.time()
-        time_stop = curr_time + 5
-
         # record until specified time
         frames = []
-        while curr_time < time_stop:
+        # while curr_time < time_stop and not self.stop_recording:
+        while not stop_recording[0]:
             data = stream.read(1024)
             frames.append(data)
-            curr_time = time.time()
 
         # Stop and close the stream
         stream.stop_stream()
         stream.close()
         p.terminate()
-
+        stop_recording[0] = False
```

Comment:
could you add a blank line number this line?

Reply:
actually I don't think we even need this line, "stop_recording[0] = False" because we check it just a few lines higher so we already know it is false.  It was required when there was still a timeout

Reply:
Hi! We have to update the state of the stop_recording back to false otherwise it would not work for the second time, as globally state would be true always so it will not enter the while loop.

Reply:
More over I can give some spaces and add a comment for the same

Reply:
yes, sorry, you're right